ui/src/components/data-source-panel.tsx

```diff
@@ -392,4 +413,23 @@ export default class DataSourcePanel extends React.Component<DataSourcePanelProp
    onSelectChange = (event) => {
       this.setState({ selectedDataSource: event })
    }
-}
\ No newline at end of file
+
+   removeDataSource = () => {
+      let connected = this.props.connectedDataSources;
+      let index = connected.indexOf(this.state.selectedDataSource.value);
+      connected.splice(index, 1);
+   }
+   
+   
+   swithcMode = () => {
```

Comment:
this function body can be:
`this.setState({editMode: !editMode})`
(the actual div should be inline just like I mentioned in the above comment)